domains/transfer/Transfer.API/API/Claiming/Api/Controllers/ClaimAutomationController.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using API.Claiming.Api.Dto.Response;
+using API.Claiming.Api.Models;
+using API.Claiming.Api.Repositories;
+using API.Shared.Extensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Claiming.Api.Controllers;
+
+[Authorize]
+[ApiController]
+[Route("api/claim-automation")]
+public class ClaimAutomationController : ControllerBase
+{
+    private readonly IClaimRepository claimRepository;
+
+    public ClaimAutomationController(IClaimRepository claimRepository)
+    {
+        this.claimRepository = claimRepository;
+    }
+
+    [HttpPost("start")]
+    [ProducesResponseType(typeof(ClaimSubject), 201)]
+    [ProducesResponseType(typeof(ClaimSubject), 200)]
+    public async Task<ActionResult> StartClaimAutomation()
+    {
+        var subject = User.FindSubjectGuidClaim();
+        var claim = await claimRepository.GetClaimSubject(Guid.Parse(subject));
+
+        if (claim != null)
+        {
+            return Ok(claim);
+        }
+
+        var claimSubject = new ClaimSubject(Guid.Parse(subject));
+
+        try
+        {
+            claim = await claimRepository.AddClaimSubject(claimSubject);
+            return CreatedAtAction(nameof(GetClaimAutomation), null, claim);
+        }
+        catch (DbUpdateException)
+        {
+            return CreatedAtAction(nameof(GetClaimAutomation), null, claimSubject);
+        }
```

Comment:
Why is this needed?

Reply:
To catch if a subject with same primary key already exists